src/cached_digest.ml

```diff
@@ -27,14 +29,49 @@ let cache = lazy (
   | None ->
     { checked_key = 0
     ; table = Path.Table.create 1024
+    ; max_timestamp = 0.
     }
   | Some cache ->
     cache.checked_key <- cache.checked_key + 1;
     cache)
 
+let get_current_filesystem_time () =
+  let special_path = Path.relative Path.build_dir ".filesystem-clock" in
+  let modify_file () =
+    let contents =
+      match Io.read_file special_path with
+      | exception Sys_error _ -> None
+      | exception Unix.Unix_error (ENOENT, _, _) -> None
+      | v -> match Int.of_string (String.trim v) with
+        | None -> None
+        | Some v -> Some v
```

Comment:
Are you missing something here? This match clause looks like an identity to me.

Reply:
I thought I needed to catch an exception, so this started out as "exception _ -> None", then changed into this and I forgot to simplify. Simplfied now. :-)